draft-ietf-mls-protocol.md

```diff
@@ -1551,14 +1541,13 @@ affect the group init_secret, it remains to be proven correct. ]]
 
 When the Application Secret is updated:
 
-- Senders MUST use the generation N+1 of the application secret, where N is
-the last generation they received.
+- Senders MUST monotonically increment the generation of their
+  Application secret.
 - Recipients SHOULD delete older generations of application secret and as soon
 as possible, within usability bounds.
 
 These rules imply that in most circumstances, an application secret will be
-used for exactly one message. Recipients MUST NOT enforce
-a one-message-per-application-secret limit.
+used for exactly one message. Note that while
```

Comment:
Why in most cases? It should never happen.

Reply:
This text generally seems like it needs work:

The rules are these:

1. Senders MUST only use each secret once
2. Receivers MUST delete an application secret once it has been used to successfully decrypt something.
3. Receivers SHOULD delete application secrets once any newer application secret has been used but MAY keep the corresponding leaf keys around for some reasonable period.

Reply:
I rewrote this part. Thanks !